docs/advanced-features/athena.ipynb

```diff
@@ -0,0 +1,688 @@
+    "The DDL below creates a view that contains package-level information including: \n",
+    "* User\n",
+    "* Package name\n",
+    "* Tophash\n",
+    "* Timestamp\n",
+    "* Commit message\n",
+    "<!--pytest-codeblocks:cont-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "id": "d8d48a9c-62a0-46d0-960a-ba938d81644d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "SLASH=r'\\/([^\\/]+)'\n",
+    "S1=r'\\/'\n",
+    "S3_MATCH=f'^s3:{S1}{SLASH}{SLASH}{SLASH}{SLASH}'\n",
+    "DDL[PACKAGES_VIEW] = f\"\"\"\n",
+    "CREATE OR REPLACE VIEW {ATHENA_DB}.{PACKAGES_VIEW} AS\n",
+    "WITH\n",
+    "  npv AS (\n",
+    "    SELECT\n",
+    "      regexp_extract(\"$path\", '{S3_MATCH}', 4) as user,\n",
+    "      regexp_extract(\"$path\", '{S3_MATCH}{SLASH}', 5) as name,\n",
+    "      regexp_extract(\"$path\", '[^/]+$') as timestamp,\n",
+    "      {PACKAGES_TABLE}.\"hash\"\n",
+    "      FROM {PACKAGES_TABLE}\n",
+    "  ),\n",
+    "  mv AS (\n",
+    "    SELECT\n",
+    "      regexp_extract(\"$path\", '[^/]+$') as tophash,\n",
+    "        manifest.\"meta\",\n",
+    "        manifest.\"message\"\n",
+    "      FROM\n",
+    "        {MANIFEST_TABLE} as manifest\n",
+    "      WHERE manifest.\"logical_key\" IS NULL\n",
+    "  )\n",
+    "SELECT\n",
+    "  npv.\"user\",\n",
+    "  npv.\"name\",\n",
+    "  npv.\"hash\",\n",
+    "  npv.\"timestamp\",\n",
+    "  mv.\"message\",\n",
+    "  mv.\"meta\"\n",
+    "FROM npv\n",
+    "JOIN\n",
+    "  mv\n",
+    "ON\n",
+    "  npv.\"hash\" = mv.\"tophash\"\n",
+    "\"\"\"\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2111bfa0-ccea-4169-aeda-b67b8b317ed9",
+   "metadata": {},
+   "source": [
+    "### D. View of object-Level metadata\n",
+    "The DDL below creates a view that contains package contents, including:\n",
+    "* logical_key\n",
+    "* physical_keys\n",
+    "* object hash\n",
+    "* object metadata\n",
+    "<!--pytest-codeblocks:cont-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "04648076-121a-488a-abd3-c8c0345e92a9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "DDL[OBJECTS_VIEW] = f\"\"\"\n",
+    "CREATE OR REPLACE VIEW {ATHENA_DB}.{OBJECTS_VIEW} AS\n",
+    "WITH\n",
+    "  mv AS (\n",
+    "    SELECT\n",
+    "      regexp_extract(\"$path\", '[^/]+$') as tophash,\n",
+    "      manifest.\"logical_key\",\n",
+    "      manifest.\"physical_keys\",\n",
+    "      manifest.\"size\",\n",
+    "      manifest.\"hash\",\n",
+    "      manifest.\"meta\",\n",
+    "      manifest.\"user_meta\"\n",
+    "    FROM\n",
+    "      {MANIFEST_TABLE} as manifest\n",
+    "    WHERE manifest.\"logical_key\" IS NOT NULL\n",
+    "  )\n",
+    "SELECT\n",
+    "  npv.\"user\",\n",
+    "  npv.\"name\",\n",
+    "  npv.\"timestamp\",\n",
+    "  mv.\"tophash\",\n",
+    "  mv.\"logical_key\",\n",
+    "  mv.\"physical_keys\",\n",
```

Comment:
In this view I think we should select a first item of array because this array always contains just a single item.

Reply:
So: `physical_keys[0] as physical_key` singular?

Reply:
Well, I think it could be done only for view. I suggested this because I thought that users are supposed to use views mostly.

Reply:
Makes sense. Fixed.